node/src/components/consensus/highway_core/highway.rs

```diff
@@ -506,6 +506,9 @@ impl<C: Context> Highway<C> {
             }
             Vertex::Endorsements(endorsements) => {
                 let unit = *endorsements.unit();
+                if endorsements.endorsers.is_empty() {
+                    return Err(EndorsementError::Empty.into());
+                }
```

Comment:
Is that necessary? If there are no equivocators we don't have to create endorsements.

Reply:
This is to defend against someone spamming us with endorsements. A peer could send us `Endorsements` vertices with _empty_ lists, and in that case we shouldn't add entries to our maps.